src/NServiceBus.RavenDB/Subscriptions/RavenDbSubscriptionSettingsExtensions.cs

```diff
@@ -73,8 +73,34 @@ public static PersistenceExtensions<RavenDBPersistence> CacheSubscriptionsFor(th
         /// <returns></returns>
         public static PersistenceExtensions<RavenDBPersistence> DisableSubscriptionVersioning(this PersistenceExtensions<RavenDBPersistence> cfg)
         {
-            cfg.GetSettings().Set(DisableSubscriptionVersioningKey, true);
+            SetLegacyVersionedSubscriptions(cfg, false);
             return cfg;
         }
+
+        /// <summary>
+        /// Continue to use legacy method of subscription versioning that includes the message assembly
+        /// major version in the subscription document lookup key. This option will be obsoleted in NServiceBus.RavenDB 6.0.0,
+        /// so subscriptions should be converted to the new format, after which the <see cref="DisableSubscriptionVersioning"/>
+        /// option should be used instead.
+        /// </summary>
+        /// <param name="cfg"></param>
+        /// <returns></returns>
+        [ObsoleteEx(TreatAsErrorFromVersion = "6.0.0", RemoveInVersion = "7.0.0")]
+        public static PersistenceExtensions<RavenDBPersistence> UseLegacyVersionedSubscriptions(this PersistenceExtensions<RavenDBPersistence> cfg)
+        {
+            SetLegacyVersionedSubscriptions(cfg, true);
+            return cfg;
+        }
+
+        static void SetLegacyVersionedSubscriptions(PersistenceExtensions<RavenDBPersistence> cfg, bool value)
+        {
+            var settings = cfg.GetSettings();
+            var current = settings.GetOrDefault<bool?>(LegacySubscriptionVersioningKey);
```

Comment:
This works, but I prefer using `settings.TryGet` instead, especially now with C# 7's out variable declaration. It composes well with making the call directly inside the `if` below.